src/crypto/crypto_keys.cc

```diff
@@ -597,35 +597,35 @@ size_t ManagedEVPPKey::size_of_public_key() const {
       pkey_.get(), nullptr, &len) == 1) ? len : 0;
 }
 
-Maybe<bool> ManagedEVPPKey::ToEncodedPublicKey(
+bool ManagedEVPPKey::ToEncodedPublicKey(
     Environment* env,
     ManagedEVPPKey key,
     const PublicKeyEncodingConfig& config,
     Local<Value>* out) {
-  if (!key) return Nothing<bool>();
+  CHECK(key);
   if (config.output_key_object_) {
     // Note that this has the downside of containing sensitive data of the
     // private key.
     std::shared_ptr<KeyObjectData> data =
           KeyObjectData::CreateAsymmetric(kKeyTypePublic, std::move(key));
-    return Just(KeyObjectHandle::Create(env, data).ToLocal(out));
+    return KeyObjectHandle::Create(env, data).ToLocal(out);
   }
-  return Just(WritePublicKey(env, key.get(), config).ToLocal(out));
+  return WritePublicKey(env, key.get(), config).ToLocal(out);
 }
 
-Maybe<bool> ManagedEVPPKey::ToEncodedPrivateKey(
+bool ManagedEVPPKey::ToEncodedPrivateKey(
     Environment* env,
     ManagedEVPPKey key,
     const PrivateKeyEncodingConfig& config,
     Local<Value>* out) {
-  if (!key) return Nothing<bool>();
+  CHECK(key);
   if (config.output_key_object_) {
     std::shared_ptr<KeyObjectData> data =
         KeyObjectData::CreateAsymmetric(kKeyTypePrivate, std::move(key));
-    return Just(KeyObjectHandle::Create(env, data).ToLocal(out));
+    return KeyObjectHandle::Create(env, data).ToLocal(out);
   }
 
-  return Just(WritePrivateKey(env, key.get(), config).ToLocal(out));
+  return WritePrivateKey(env, key.get(), config).ToLocal(out);
```

Comment:
This method *should* have a `Maybe<bool>` return type, because `WritePrivateKey` can throw an exception. The mistake here was using `Just()` unconditionally previously, when it should have forwarded an empty `Maybe` result instead.